pkg/storage/etcd3/watcher.go

```diff
@@ -109,6 +107,10 @@ func (wc *watchChan) run() {
 
 	select {
 	case err := <-wc.errChan:
+		if err == context.Canceled {
+			wc.cancel() // just in case
```

Comment:
I'm slightly confused why you would cancel the context when it already came back as canceled?